scripts/auto-stop-idle/autostop.py

```diff
@@ -0,0 +1,110 @@
+#     Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#     Licensed under the Apache License, Version 2.0 (the "License").
+#     You may not use this file except in compliance with the License.
+#     A copy of the License is located at
+#
+#         https://aws.amazon.com/apache-2-0/
+#
+#     or in the "license" file accompanying this file. This file is distributed
+#     on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+#     express or implied. See the License for the specific language governing
+#     permissions and limitations under the License.
+
+import requests
+from datetime import datetime
+import getopt, sys, re
+import urllib3
+import boto3
+
+urllib3.disable_warnings(urllib3.excepqqtions.InsecureRequestWarning)
+
+# Usage
+usageInfo = """Usage:
+This scripts checks if a notebook is idle for X seconds if it does, it'll stop the notebook:
+python autostop.py --time <time_in_seconds> [--port <jupyter_port>] [--ignore-connections]
+Type "python autostop.py -h" for available options.
+"""
+# Help info
+helpInfo = """-t, --time
+    Auto stop time in seconds
+-p, --port
+    jupyter port
+-c --ignore-connections
+    Stop notebook once idle, ignore connected users
+-h, --help
+    Help information
+"""
+
+# Read in command-line parameters
+idle = True
+port = '8443'
+ignore_connections = False
+try:
+    opts, args = getopt.getopt(sys.argv[1:], "ht:p:c", ["help","time=","port=","ignore-connections"])
+    if len(opts) == 0:
+        raise getopt.GetoptError("No input parameters!")
+    for opt, arg in opts:
+        if opt in ("-h", "--help"):
+            print(helpInfo)
+            exit(0)
+        if opt in ("-t", "--time"):
+            time = int(arg)
+        if opt in ("-p", "--port"):
+            port = str(arg)
+        if opt in ("-c", "--ignore-connections"):
+            ignore_connections = True
+except getopt.GetoptError:
+    print(usageInfo)
+    exit(1)
+
+# Missing configuration notification
+missingConfiguration = False
+if not time:
+    print("Missing '-t' or '--time'")
+    missingConfiguration = True
+if missingConfiguration:
+    exit(2)
+
+
+def is_idle(last_activity):
+    last_activity = datetime.strptime(last_activity,"%Y-%m-%dT%H:%M:%S.%fz")
+    if (datetime.now() - last_activity).total_seconds() > time:
+        return True
+    else:
+        return False
+
+
+def get_notebook_name():
```

Comment:
The notebook instance name should be fetched via the metadata file present on the instance rather than log parsing.
https://docs.aws.amazon.com/sagemaker/latest/dg/nbi-metadata.html

Reply:
Thanks, wasn't aware of that, updated to use the nbi-metadata.